lnd_test.go

```diff
@@ -1886,6 +1886,47 @@ out:
 	}
 }
 
+// testNodeAnnouncement ensures that when a node is started with one or more
+// external IP addresses specified on the command line, that those addresses
+// announced to the network and reported in the network graph.
+func testNodeAnnouncement(net *networkHarness, t *harnessTest) {
+	timeout := time.Duration(time.Second * 15)
+	ctxb := context.Background()
+	ctxt, _ := context.WithTimeout(ctxb, timeout)
+
+	lndArgs := []string{"--externalip=192.168.1.1:8333",
+		"--externalip=[2001:db8:85a3:8d3:1319:8a2e:370:7348]:80",
+		"--externalip=127.0.0.1:8335"}
+	dave, err := net.NewNode(lndArgs)
+	if err != nil {
+		t.Fatalf("unable to create new nodes: %v", err)
+	}
+
+	if err := net.ConnectNodes(ctxb, net.Alice, dave); err != nil {
+		t.Fatalf("unable to connect bob to carol: %v", err)
+	}
+
+	chanAmt := btcutil.Amount(btcutil.SatoshiPerBitcoin / 2)
+	chanPointAlice := openChannelAndAssert(ctxt, t, net, net.Alice, dave,
```

Comment:
Since I was using DescribeGraph to check the contents of the NodeAnnouncement, I had to open a channel here, I believe.